src/Distribution/Server/Features/Upload/State.hs

```diff
@@ -1,4 +1,4 @@
-{-# LANGUAGE DeriveDataTypeable, TypeFamilies, TemplateHaskell #-}
+{-# LANGUAGE DeriveDataTypeable, RecordWildCards, TypeFamilies, TemplateHaskell #-}
```

Comment:
I think adding this reverse index is a rather heavyweight and unnecessary change. Fetching all packages for a user is already sufficiently efficient. Cf https://hackage.haskell.org/user/BryanOSullivan -- fast!

Reply:
OK! I will change that then.

Reply:
I updated the PR using the GroupIndex mechanics and reverted all the changes to the Upload feature.